src/coreclr/src/vm/genericdict.cpp

```diff
@@ -1245,6 +1495,13 @@ Dictionary::PopulateEntry(
 
         if ((slotIndex != 0) && !IsCompilationProcess())
         {
+            // Lock is needed because dictionary pointers can get updated during dictionary size expansion
+            CrstHolder ch(&SystemDomain::SystemModule()->m_DictionaryCrst);
+
+            Dictionary* pDictionary = (pMT != NULL) ?
```

Comment:
This may be growing the dictionaries unnecessarily. Should this check whether the dictionary that we got is big enough for slotIndex first, and grow the dictionary only when it is not?

Reply:
Thanks Jan, yes that's a good point. We can defer expansion until we hit a slot that is actually beyond the size of the dictionary. I'll make that update.